tests/integration_tests/test_mlflow.py

```diff
@@ -0,0 +1,160 @@
+    y = trial.suggest_loguniform("y", 20, 30)
+    z = trial.suggest_categorical("z", (-1.0, 1.0))
+    assert isinstance(z, float)
+    return (x - 2) ** 2 + (y - 25) ** 2 + z
+
+
+def test_happy_case(tmpdir):
+    # type: (py.path.local) -> None
+
+    tracking_file_name = "file:{}".format(tmpdir)
+
+    mlflc = MLflowCallback(
+        tracking_uri=tracking_file_name, metric_name="my_metric", experiment="my_experiment"
+    )
+    study = optuna.create_study()
+    ft = _create_frozen_trial()
+    mlflc(study, ft)
+
+
+@patch("mlflow.set_tags")
+def test_set_tags(set_tags, tmpdir):
+    # type: (MagicMock, py.path.local) -> None
+
+    tracking_file_name = "file:{}".format(tmpdir)
+
+    mlflc = MLflowCallback(
+        tracking_uri=tracking_file_name, metric_name="my_metric", experiment="my_experiment"
+    )
+    study = optuna.create_study()
+    ft = _create_frozen_trial()
+    mlflc(study, ft)
+
+    assert set_tags.called
+    assert set_tags.call_count == 1
+    call_arg = set_tags.call_args_list[0][0][0]
+    assert call_arg["trial_state"] == "TrialState.COMPLETE"
+    assert call_arg["x"] == "UniformDistribution(high=12, low=5)"
+    assert call_arg["trial_number"] == "0"
+
+
+@patch("mlflow.log_params")
+def test_log_params(log_params, tmpdir):
+    # type: (MagicMock, py.path.local) -> None
+
+    tracking_file_name = "file:{}".format(tmpdir)
+
+    mlflc = MLflowCallback(
+        tracking_uri=tracking_file_name, metric_name="my_metric", experiment="my_experiment"
+    )
+    study = optuna.create_study()
+    ft = _create_frozen_trial()
+    mlflc(study, ft)
+
+    assert log_params.called
+    assert log_params.call_count == 1
+    call_arg = log_params.call_args_list[0][0][0]
+    assert call_arg["x"] == 10
+
+
+@patch("mlflow.log_metric")
+def test_log_metric_with_metric_name(log_metric, tmpdir):
+    # type: (MagicMock, py.path.local) -> None
+
+    tracking_file_name = "file:{}".format(tmpdir)
+
+    mlflc = MLflowCallback(
+        tracking_uri=tracking_file_name, metric_name="my_metric", experiment="my_experiment"
+    )
+    study = optuna.create_study()
+    ft = _create_frozen_trial()
+    mlflc(study, ft)
+
+    assert log_metric.called
+    assert log_metric.call_count == 1
+    call_args = log_metric.call_args_list[0][0]
+    assert call_args[0] == "my_metric"
+    assert call_args[1] == 0.2
+
+
+@patch("mlflow.log_metric")
+def test_log_metric_with_default_metric_name(log_metric, tmpdir):
+    # type: (MagicMock, py.path.local) -> None
+
+    tracking_file_name = "file:{}".format(tmpdir)
+
+    mlflc = MLflowCallback(tracking_uri=tracking_file_name, experiment="my_experiment")
+    study = optuna.create_study()
+    ft = _create_frozen_trial()
+    mlflc(study, ft)
+
+    assert log_metric.called
+    assert log_metric.call_count == 1
+    call_args = log_metric.call_args_list[0][0]
+    assert call_args[0] == "trial_value"
+    assert call_args[1] == 0.2
+
+
+@patch("mlflow.log_metric")
```

Comment:
I think we no longer need mock for those tests, because we're currently able to put the data to the temporary directory instead of an external MLflow service. So, removing mock would make the test cases more practical.

What do you think?

Reply:
Yes that's right. I removed the mocks and replaced them by loading the experiments and runs with the MLFlow client.